pkgs/development/libraries/asio/generic.nix

```
@@ -21,7 +21,7 @@ stdenv.mkDerivation {
     homepage = "http://asio.sourceforge.net/";
     description = "Cross-platform C++ library for network and low-level I/O programming";
     license = licenses.boost;
-    broken = stdenv.isDarwin;  # test when updating to >=1.12.1
+    broken = stdenv.isDarwin && version < "1.16.1";
```

```suggestion
    broken = stdenv.isDarwin && stdenv.lib.versionOlder version "1.16.1";
```
otherwise it's doing a simple string comparison:
```
nix-repl> lib.versionOlder "1.2" "1.11"
true
nix-repl> "1.2" < "1.11"
false
```

---

Ooh, thank you!
I learned something with that